spec/requests/users_spec.rb

```diff
@@ -0,0 +1,38 @@
+require 'rails_helper'
+
+RSpec.describe 'Posts Controller', type: :request do
+  it 'GET /users should succeed' do
+    get '/users'
+    expect(response).to have_http_status(:ok)
+  end
+
+  it "Redirects to the users' list page" do
+    get '/users/id/posts'
+    expect(response).to render_template(:index)
+    expect(response.body).to include('User name')
+    expect(response.body).to include('Number of posts: x')
+  end
+
+  it 'does not render a different template' do
+    get '/users'
+    expect(response).to_not render_template(:show)
+  end
+
+  it 'GET users/id should succeed' do
+    get '/users/id'
+    expect(response).to have_http_status(:ok)
+  end
+
+  it 'Redirects to the user profile page' do
+    get '/users/id/'
+    expect(response).to render_template(:show)
```

Comment:
- [ ] On line no.3, change the name to `Users controller` since this is the test for users.
- [ ] Same here. I would recommend wrapping the index method test under one `describe` block and another `describe` block for show.

```ruby
RSpec.describe 'Users Controller', type: :request do
  describe 'GET /index' do
    ...
  end
 
  describe 'GET /show' do
    ...
  end
end

```

Reply:
Hey @Deepakdanger

Thank You for the feedback :pray: 
I really appreciate :+1: :+1: